mutable_buffer/src/chunk.rs

```diff
@@ -152,6 +167,53 @@ impl Chunk {
         }
     }
 
+    pub fn is_open(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Open
+    }
+
+    pub fn is_closing(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Closing
+    }
+
+    pub fn is_closed(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Closed
+    }
+
+    pub fn is_moving(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Moving
+    }
+
+    pub fn is_moved(&self) -> bool {
+        let state = self.state.lock();
+        *state == ChunkState::Moved
+    }
+
+    pub fn same_state(&self, chunk_state: ChunkState) -> bool {
+        match chunk_state {
+            ChunkState::Open => self.is_open(),
+            ChunkState::Closing => self.is_closing(),
+            ChunkState::Closed => self.is_closed(),
+            ChunkState::Moving => self.is_moving(),
+            ChunkState::Moved => self.is_moved(),
+        }
```

Comment:
```suggestion
    pub fn same_state(&self, chunk_state: ChunkState) -> bool {
        chunk_state == *self.state.lock()
    }
```
I think you could avoid some of the repetition if you wanted to

Reply:
Right. Very silly code lines :)